include/boost/http_proto/fields_base.hpp

```diff
@@ -397,36 +397,40 @@ class BOOST_SYMBOL_VISIBLE
 
     /** Set a header value
 
-        This sets the value of the header
-        at `it`. The name is not changed.
-        <br>
-        No iterators are invalidated.
+        This function uses the given value to overwrite the
+        current one in the header field pointed to by the
+        supplied iterator. The value must be syntactically
+        valid or else an error is returned.
+        Any leading or trailing whitespace in the new value
+        is ignored.
 
         @par Complexity
 
         @par Exception Safety
         Strong guarantee.
         Calls to allocate may throw.
 
+        @return The error, if any occurred.
+
         @param it An iterator to the header.
 
         @param value A value, which
-        @li Must be syntactically valid for the header,
-        @li Must be semantically valid for the message, and
-        @li May not contain leading or trailing whitespace.
+        @li Must be semantically valid for the message.
     */
     BOOST_HTTP_PROTO_DECL
-    void
+    system::result<void>
     set(
         iterator it,
         core::string_view value);
 
     /** Set a header value
 
-        This function sets the value of the
-        header with the specified field id.
-        Other headers with the same name
-        are removed first.
+        This function modifies the container to contain exactly
+        one field with the specified id set to the given value,
```

Comment:
When you get tired of saying "This function" you might switch it up to "The container is modified to contain..." just to keep the reader interested.